pkg/database/database.go

```diff
@@ -5,408 +5,103 @@ package database
 
 import (
 	"context"
-	"database/sql"
-	"encoding/json"
 	"errors"
 	"fmt"
 	"log/slog"
-	"slices"
-	"strings"
+	"sync"
 
 	"github.com/huandu/go-sqlbuilder"
 	"github.com/jmoiron/sqlx"
 	"github.com/kubearchive/kubearchive/pkg/database/facade"
-	"github.com/kubearchive/kubearchive/pkg/models"
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
 const defaultContainerAnnotation = "kubectl.kubernetes.io/default-container"
 
 var ResourceNotFoundError = errors.New("resource not found")
 
-type newDatabaseFunc func(*sqlx.DB) DBInterface
 type newDBCreatorFunc func(map[string]string) facade.DBCreator
 
-var RegisteredDatabases = make(map[string]newDatabaseFunc)
+var RegisteredDatabases = make(map[string]Database)
 var RegisteredDBCreators = make(map[string]newDBCreatorFunc)
 
-type DBInterface interface {
-	QueryResources(ctx context.Context, kind, apiVersion, namespace,
-		name, continueId, continueDate string, labelFilters *LabelFilters, limit int) ([]string, int64, string, error)
-	QueryLogURL(ctx context.Context, kind, apiVersion, namespace, name string) (string, string, error)
-
-	WriteResource(ctx context.Context, k8sObj *unstructured.Unstructured, data []byte) error
-	WriteUrls(ctx context.Context, k8sObj *unstructured.Unstructured, jsonPath string, logs ...models.LogTuple) error
-	Ping(ctx context.Context) error
-	CloseDB() error
+type Database interface {
+	DBReader
+	DBWriter
 }
 
-type Database struct {
-	DB       *sqlx.DB
-	Flavor   sqlbuilder.Flavor
-	Selector facade.DBSelector
-	Filter   facade.DBFilter
-	Sorter   facade.DBSorter
-	Inserter facade.DBInserter
-	Deleter  facade.DBDeleter
+type DatabaseImpl struct {
```

Comment:
does `DatabaseImpl` need to be exported from the database package? If the sink is using `DBWriter` and the api is using `DBReader`, it seems like we could leave `DatabaseImpl` type unexported

```suggestion
type databaseImpl struct {
```

Reply:
Fixed! Note https://github.com/kubearchive/kubearchive/pull/921#issuecomment-2741147390